### PR TITLE
feat(medium-pass-1): use getProvider and getAssessmentActionMessageCreator in deps instead of using classes directly

### PR DESCRIPTION
**Pull request by madalynrose**
#### Details

This is a follow-up PR to https://github.com/microsoft/accessibility-insights-web/pull/6236. This covers some of the changes for `deps.assessmentsProvider` and all of the changes for `deps.assessmentActionMessageCreator`.

##### Motivation

feature work 🚀 

##### Context

This is a continuation of the work done in #6240

#### Pull request checklist

- [n/a] Addresses an existing issue: #0000
- [x] Ran `yarn null:autoadd`
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [n/a] (UI changes only) Added screenshots/GIFs to description above
- [n/a] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/DetailsView/components/details-view-command-bar.tsx [hunk 1]
```diff
@@ -55,7 +55,7 @@ export type DetailsViewCommandBarDeps = {
     reportGenerator: ReportGenerator;
     getDateFromTimestamp: (timestamp: string) => Date;
     detailsViewActionMessageCreator: DetailsViewActionMessageCreator;
-    assessmentActionMessageCreator: AssessmentActionMessageCreator;
+    getAssessmentActionMessageCreator: () => AssessmentActionMessageCreator;
```
Comment (waabid) [inline src/DetailsView/components/details-view-command-bar.tsx:58]: nit: It seems like we're not using this here; we should probably remove it entirely.

File: src/DetailsView/components/details-view-overlay/details-view-overlay.tsx [hunk 2]
```diff
@@ -24,7 +24,7 @@ import {
 
 export type DetailsViewOverlayDeps = {
     detailsViewActionMessageCreator: DetailsViewActionMessageCreator;
-    assessmentActionMessageCreator: AssessmentActionMessageCreator;
+    getAssessmentActionMessageCreator: () => AssessmentActionMessageCreator;
```
Comment (waabid) [inline src/DetailsView/components/details-view-overlay/details-view-overlay.tsx:27]: nit: it seems like we're not using this here; we should probably remove it entirely.

File: src/DetailsView/components/requirement-view.tsx [hunk 3]
```diff
@@ -30,7 +30,7 @@ import * as React from 'react';
 import styles from './requirement-view.scss';
 
 export type RequirementViewDeps = {
-    assessmentActionMessageCreator: AssessmentActionMessageCreator;
+    getAssessmentActionMessageCreator: () => AssessmentActionMessageCreator;
```
Comment (waabid) [inline src/DetailsView/components/requirement-view.tsx:33]: nit: it seems like we're not using this here; we should probably remove it entirely.

File: src/DetailsView/components/test-view-container.tsx [hunk 4]
```diff
@@ -39,7 +39,7 @@ import { TestViewDeps } from './test-view';
 
 export type TestViewContainerDeps = {
     detailsViewActionMessageCreator: DetailsViewActionMessageCreator;
-    assessmentActionMessageCreator: AssessmentActionMessageCreator;
+    getAssessmentActionMessageCreator: () => AssessmentActionMessageCreator;
```
Comment (waabid) [inline src/DetailsView/components/test-view-container.tsx:42]: nit: it seems like we're not using this here; we should probably remove it entirely.

**Review by waabid — APPROVED** [review]
LGTM! Just a few places where we can remove the dep entirely.

**Review by lisli1 — APPROVED** [review]

**Pull request merged**